giddy/directional.py

```diff
@@ -328,16 +330,30 @@ def plot(self, attribute=None):
         ----------
         attribute : (n,) ndarray, optional
             Variable to specify colors of the colorbars.
+
         """
-        import matplotlib.cm as cm
-        import matplotlib.pyplot as plt
-        ax = plt.subplot(111, projection='polar')
-        ax.set_rlabel_position(315)
-        if attribute is None:
-            c = ax.scatter(self.theta, self.r)
+        use_splot = False
+        try:
+            import splot.giddy
+            use_splot = True
+        except ImportError:
+            warnings.warn('To get better results, please install the `splot` package',
+                          UserWarning)
```

Comment:
*If* the final goal is to remove the custom code here below (for case of `use_splot = False`), I would give a stronger message (and make it a DeprecationWarning) making it clear in the future having splot installed will be required.